rllib/core/rl_module/rl_module.py

```diff
@@ -33,50 +34,113 @@ class SingleAgentRLModuleSpec:
+        if self.model_config_dict is None:
+            raise ValueError("Model config is not set.")
 
-        return self.module_class.from_model_config(
-            observation_space=self.observation_space,
-            action_space=self.action_space,
-            model_config_dict=self.model_config,
+        module_config = self.get_rl_module_config()
+        return self.module_class(module_config)
+
+    @classmethod
+    def from_module(cls, module: "RLModule") -> "SingleAgentRLModuleSpec":
+        from ray.rllib.core.rl_module.marl_module import MultiAgentRLModule
+
+        if isinstance(module, MultiAgentRLModule):
+            raise ValueError(
+                "MultiAgentRLModule cannot be converted to SingleAgentRLModuleSpec."
+            )
+
+        return SingleAgentRLModuleSpec(
+            module_class=type(module),
+            observation_space=module.config.observation_space,
+            action_space=module.config.action_space,
+            model_config_dict=module.config.model_config_dict,
+        )
+
+    def to_dict(self):
+        """Returns a serialized representation of the spec."""
+
+        return {
+            "module_class": self.module_class,
+            "module_config": self.get_rl_module_config().to_dict(),
+        }
+
+    @classmethod
+    def from_dict(cls, d):
+        """Returns a single agent RLModule spec from a serialized representation."""
+        module_class = d["module_class"]
+
+        module_config = RLModuleConfig.from_dict(d["module_config"])
+        observation_space = module_config.observation_space
+        action_space = module_config.action_space
+        model_config_dict = module_config.model_config_dict
+
+        return SingleAgentRLModuleSpec(
+            module_class=module_class,
+            observation_space=observation_space,
+            action_space=action_space,
+            model_config_dict=model_config_dict,
         )
 
 
 @ExperimentalAPI
 @dataclass
 class RLModuleConfig:
-    """Configuration for the PPO module.
-    # TODO (Kourosh): Whether we need this or not really depends on how the catalog
-    # design end up being.
-    Attributes:
-        observation_space: The observation space of the environment.
-        action_space: The action space of the environment.
-        max_seq_len: Max seq len for training an RNN model.
-        (TODO (Kourosh) having max_seq_len here seems a bit unnatural, can we rethink
-        this design?)
-    """
 
     observation_space: gym.Space = None
     action_space: gym.Space = None
-    max_seq_len: int = None
+    model_config_dict: Mapping[str, Any] = None
+    catalog_class: Type["Catalog"] = None
+
+    def get_catalog(self) -> "Catalog":
+        """Returns the catalog for this config."""
+        return self.catalog_class(
+            observation_space=self.observation_space,
+            action_space=self.action_space,
+            model_config_dict=self.model_config_dict,
```

Comment:
At some point we'll need the view_requirements here as well.
Do you think they should be part of RLModuleConfig?
Not saying we should include it in this PR.

Reply:
view_requirements will probably be a property of RLModules and passed to `get_catalog(vr)` to get the catalog. We'll decide this when we get to it. Another way is to make it part of the config. Idk at this point.